sqlx-core/src/rt/mod.rs

```diff
@@ -48,13 +57,15 @@ pub async fn sleep(duration: Duration) {
         return tokio::time::sleep(duration).await;
     }
 
-    #[cfg(feature = "_rt-async-std")]
-    {
-        async_std::task::sleep(duration).await
+    cfg_if! {
+        if #[cfg(feature = "_rt-async-std")] {
+            async_std::task::sleep(duration).await
+        } else if #[cfg(feature = "_rt-smol")] {
+            rt_smol::sleep(duration).await
```

Comment:
Sleeping is also backed by `async_io::Timer` in both cases:

* https://docs.rs/async-std/latest/src/async_std/utils.rs.html#66
* `smol::Timer` _is_ just `async_io::Timer`: https://docs.rs/smol/latest/src/smol/lib.rs.html#52